File: service_java/src/main/java/com/coldlt/crud/api/model/Factura.java
package com.coldlt.crud.api.model;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;
import javax.persistence.Column;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.Lob;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import org.springframework.stereotype.Component;

/**
 *
 * @author ANDRES
 */
@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Table(name = "FACTURA")
public class Factura implements Serializable {

    @Id
    @GeneratedValue
    private Long id;
    private String cliente;
    @Lob
    private ArrayList<Item> items;
    private int valorTotal;

}

File: client_angular/src/app/factura.ts
import { Item } from './item';

export class Factura {

    cliente: string
    items:  Array<Item> = [];
    valorTotal: number
    constructor(
        cliente1: string,
        items1:   Array<Item>,
        valorTotal1: number
    ) {
        this.cliente = cliente1
        this.items =  items1
        this.valorTotal = valorTotal1
    }
}